doc/README.txt

```diff
@@ -72,6 +72,17 @@ patoolib.search_archive("def urlopen", "python3.3.tar.gz")
 patoolib.repack_archive("linux-2.6.33.tar.gz", "linux-2.6.33.tar.bz2")
 ```
 
+Bash completion
+----------------
+To enable bash completion
+- install python `argcomplete` package,
+- run `activate-global-python-argcomplete` to create `/etc/bash_completion.d/python-argcomplete.sh`,
+- add `eval "$(register-python-argcomplete patool)"` to your `~/.bashrc`.
+
+Open new bash session.
+
+More information about `argcomplete` installation and usage can be found [here](https://github.com/kislyuk/argcomplete).
```

Comment:
FWIW in datalad (and other projects) we just added a command such as 
```
$> datalad shell-completion --help-np
Usage: datalad shell-completion [-h] [--version]

Display shell script for enabling shell completion for DataLad.

Output of this command should be "sourced" by the bash or zsh to enable
shell completions provided by argcomplete.

Example:

    $ source <(datalad shell-completion)
    $ datalad --<PRESS TAB to display available option>

optional arguments:
  -h, --help, --help-np
                        show this help message. --help-np forcefully disables the use of a pager for displaying the help
                        message
  --version             show the module and its version which provides the command
```
so users (and myself) doesn't need to RTFM and can find/spot such command among available in the tool CLI

Reply:
It's was long time ago - can't catch up detail now. I'd like to suggest closing this request as outdated.